cli/running/running.go

```diff
@@ -324,88 +445,149 @@ func createPIDFile(pidFileName string) error {
 	return nil
 }
 
+// makePath make application path with rules:
+// * if path is not set:
+//     * if single instance application: baseBath + application name.
+//     * else : baseBath + application name + instance name.
+// * if path is set and it is absolute:
+//    * if single instance application: path + application name
+//    * else: path + application name + instance name.
+// * if path is set and it is relative:
+//    * if single instance application: basePath + path + application name.
+//    * else: basePath + path + application name + instance name.
+func makePath(path string, basePath string, inst *cmdcontext.RunningCtx) string {
+	res := ""
+
+	if path == "" {
+		if inst.SingleApp {
+			return filepath.Join(basePath, inst.AppName)
+		} else {
+			res = filepath.Join(basePath, inst.AppName)
+			return filepath.Join(res, inst.InstName)
+		}
+	}
+
+	if filepath.IsAbs(path) {
+		if inst.SingleApp {
+			return filepath.Join(path, inst.AppName)
+		} else {
+			res = filepath.Join(path, inst.AppName)
+			return filepath.Join(res, inst.InstName)
+		}
+	}
+
+	res = filepath.Join(basePath, path)
+	res = filepath.Join(res, inst.AppName)
+
+	if !inst.SingleApp {
+		return filepath.Join(res, inst.InstName)
+	}
+
+	return res
+}
+
 // FillCtx fills the RunningCtx context.
 func FillCtx(cliOpts *config.CliOpts, cmdCtx *cmdcontext.CmdCtx,
 	args []string) error {
+	var err error
```

Comment:
What for? The error is created as a local variable everywhere in this function.

Reply:
This needs to prevent shadowing `basePath` in this part:
```go
    if basePath == "" {                                                                                   
        if basePath, err = os.Getwd(); err != nil {                                                       
            return fmt.Errorf(`Can't get the "RunDir: %s"`, err)                                          
        }                                                                                                 
    } 
    ```